File: bindings/kotlin/examples/CustomQuery.kt
// Copyright (c) 2025 IOTA Stiftung
// SPDX-License-Identifier: Apache-2.0

import iota_sdk.GraphQlClient
import iota_sdk.Query
import kotlinx.coroutines.runBlocking
import kotlinx.serialization.encodeToString
import kotlinx.serialization.json.Json

fun main() = runBlocking {
    val client = GraphQlClient.newDevnet()

    val queryEpochDataStr =
            """
        query MyQuery(${'$'}id: UInt53) {
            epoch(id: ${'$'}id) {
                epochId
                referenceGasPrice
                totalGasFees
                totalCheckpoints
                totalTransactions
            }
        }
    """.trimIndent()

    val queryEpochData = Query(queryEpochDataStr)
    val res1 = client.runQuery(queryEpochData)
    println(res1)

    val variables = mapOf("id" to 1)
    val queryEpochDataWithVariables = Query(queryEpochDataStr, Json.encodeToString(variables))

    val res2 = client.runQuery(queryEpochDataWithVariables)
    println(res2)

    val queryChainIdStr =
            """
        query MyQuery {
            chainIdentifier
        }
    """.trimIndent()
    val queryChainId = Query(queryChainIdStr)
    val res3 = client.runQuery(queryChainId)
    println(res3)
}

File: bindings/swift/examples/CustomQuery.swift
// Copyright (c) 2026 IOTA Stiftung
// SPDX-License-Identifier: Apache-2.0

import IotaSDK

@main
struct CustomQueryExample {
  static func main() async throws {
    let client = GraphQlClient.newTestnet()

    let queryEpochDataStr = """
      query MyQuery($id: UInt53) {
          epoch(id: $id) {
              epochId
              referenceGasPrice
              totalGasFees
              totalCheckpoints
              totalTransactions
          }
      }
      """
    let queryEpochData = Query(queryText: queryEpochDataStr)
    let res = try await client.runQuery(query: queryEpochData)
    print(res)

    let variables = "{\"id\": 1}"
    let queryEpochDataWithVariables = Query(queryText: queryEpochDataStr, variables: variables)
    let res2 = try await client.runQuery(query: queryEpochDataWithVariables)
    print(res2)

    let queryChainIdStr = """
      query MyQuery {
          chainIdentifier
      }
      """
    let queryChainId = Query(queryText: queryChainIdStr)
    let res3 = try await client.runQuery(query: queryChainId)
    print(res3)
  }
}
